src/app/services/chat.service.ts

```diff
@@ -7,19 +7,61 @@ import { ChatGroupI } from '../models/chat-group';
   providedIn: 'root',
 })
 export class ChatService {
-  constructor(private afs: AngularFirestore, private fbSvc: FirebaseService) {}
+  constructor(private afs: AngularFirestore, private fbSvc: FirebaseService) {
+    this.pairChatColQuery(
+      'da4RmsE4fUX3TjXvyUqzlxFW19a2',
+      'P9X37J3EXQNz4b8WZPJWTIR81eN2',
+    )
+      .valueChanges()
+      .subscribe(console.log);
+  }
 
-  createChat = async (participants: string[]) => {
+  createPairChat = async (uid1: string, uid2: string) => {
+    const existingChats = await this.getPairChat(uid1, uid2);
+    if (existingChats.length > 0) {
+      const existingChat = existingChats[0];
+      const { id } = existingChat;
+      return id;
+    }
+    const participantsMap = {
+      [uid1]: true,
+      [uid2]: true,
+    };
     const chatId = this.afs.createId();
     const chatGroup: ChatGroupI = {
-      participantIds: participants,
+      participantIds: participantsMap,
       createdAt: this.fbSvc.fsTimestamp(),
     };
-    await this.afs.collection('chats').doc(chatId).set(chatGroup);
+    await this.pairChatsCol().doc(chatId).set(chatGroup);
```

Comment:
```suggestion
    await this.pairChatDoc(chatId).set(chatGroup);
```
Hopefully we can now use this instead of re-creating the logic up here.